catalog/sql/sql.go

```diff
@@ -809,3 +823,350 @@ func (c *Catalog) UpdateNamespaceProperties(ctx context.Context, namespace table
+func (c *Catalog) CheckViewExists(ctx context.Context, identifier table.Identifier) (bool, error) {
+	ns := strings.Join(catalog.NamespaceFromIdent(identifier), ".")
+	viewName := catalog.TableNameFromIdent(identifier)
+
+	return withReadTx(ctx, c.db, func(ctx context.Context, tx bun.Tx) (bool, error) {
+		exists, err := tx.NewSelect().Model(&sqlIcebergTable{
+			CatalogName:    c.name,
+			TableNamespace: ns,
+			TableName:      viewName,
+		}).WherePK().Where("iceberg_type = ?", ViewType).Exists(ctx)
+		if err != nil {
+			return false, fmt.Errorf("error checking view existence: %w", err)
+		}
+
+		return exists, nil
+	})
+}
+
+// LoadView loads a view from the catalog.
+func (c *Catalog) LoadView(ctx context.Context, identifier table.Identifier) (map[string]interface{}, error) {
+	ns := strings.Join(catalog.NamespaceFromIdent(identifier), ".")
+	viewName := catalog.TableNameFromIdent(identifier)
+
+	view, err := withReadTx(ctx, c.db, func(ctx context.Context, tx bun.Tx) (*sqlIcebergTable, error) {
+		v := new(sqlIcebergTable)
+		err := tx.NewSelect().Model(v).
+			Where("catalog_name = ?", c.name).
+			Where("table_namespace = ?", ns).
+			Where("table_name = ?", viewName).
+			Where("iceberg_type = ?", ViewType).
+			Scan(ctx)
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, fmt.Errorf("%w: %s", catalog.ErrNoSuchView, identifier)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error encountered loading view %s: %w", identifier, err)
+		}
+
+		return v, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if !view.MetadataLocation.Valid {
+		return nil, fmt.Errorf("%w: %s, metadata location is missing", catalog.ErrNoSuchView, identifier)
+	}
+
+	viewMetadata := map[string]interface{}{
+		"name":              viewName,
+		"namespace":         ns,
+		"metadata-location": view.MetadataLocation.String,
+	}
+
+	fs, err := io.LoadFS(ctx, c.props, view.MetadataLocation.String)
+	if err != nil {
+		return nil, fmt.Errorf("error loading view metadata: %w", err)
+	}
+	inputFile, err := fs.Open(view.MetadataLocation.String)
+	if err != nil {
+		return viewMetadata, fmt.Errorf("error encountered loading view metadata %s: %w", identifier, err)
+	}
+	defer inputFile.Close()
+
+	var fullViewMetadata map[string]interface{}
+	if err := json.NewDecoder(inputFile).Decode(&fullViewMetadata); err != nil {
+		return viewMetadata, fmt.Errorf("error encountered decoding view metadata %s: %w", identifier, err)
+	}
+
+	fullViewMetadata["name"] = viewName
+	fullViewMetadata["namespace"] = ns
+	fullViewMetadata["metadata-location"] = view.MetadataLocation.String
+
+	if props, ok := fullViewMetadata["properties"].(map[string]interface{}); ok {
+		strProps := make(map[string]string)
+		for k, v := range props {
+			if str, ok := v.(string); ok {
+				strProps[k] = str
+			} else if vJson, err := json.Marshal(v); err == nil {
+				strProps[k] = string(vJson)
+			}
+		}
+		fullViewMetadata["properties"] = strProps
+	}
```

Comment:
same with this section, let's pull it out into a helper function that the other catalogs would be able to leverage too. We already have an internal package for shared utilities with catalogs that this could get added to.